docs/adr/006_cups.md

```diff
@@ -0,0 +1,297 @@
+
+There is no need to copy/paste manually these files in the Basic Station; the CUPS Protocol Implementation later described in this document will retrieve the needed credentials from a centralized repository and update the Basic Station configuration when it starts. This option is also providing an option to rotate certificates when these are near the expiration.
+
+### Where to store certificates
+
+There are multiple options of where to store the `cups.{trust,cert,key}` and `tc.{trust,cert,key}` bundle files.
+
+#### a. Azure Storage Account
+
+This is the easiest option for a new user of the Starter Kit, even though it's not the safest one.
+
+When deploying the Starter Kit, an Azure Storage Account is provisioned for Azure Functions.
+
+LoRa Device Provisioning tool should be capable of uploading the CLI provided bundle files and properly update the twin for the concentrator device being created in IoT Hub.
+
+When using this option, the 'Facade' Azure Function will retrieve certificate files from such storage account.
+
+#### b. Azure Key Vault secret
+
+This option requires more manual intervention, as the LoRa Device Provisioning tool is not adapted for uploading the credential bundle files to an Azure Key Vault instance.
+
+Following the documentation on [how to set a secret in Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-portal), you can just create two secrets and retrieve their URLs.
+
+When setting a Key Vault secret in the concentrator twin "cups" section, the 'Facade' Azure Function will retrieve it using a Managed Identity.
+
+#### c. External HTTPS repository
+
+This option requires most manual intervention.
+
+LoRa Device Provisioning tool is not uploading the credential bundle files to the repository.
+
+'Facade' Azure Function can handle a HTTPS endpoint from where the bundles will be downloaded, even though there is no security check or additional AuthN/AuthZ option provided here.
+
+## CUPS Protocol Implementation
+
+### Flow diagram for credential management scenarios
+
+#### Prerequisites
+
+- LoRa Basics™ Station is configured with a cups.uri, a cups.trust and mandatory cups.cert/key
+- The concentrator device twin should include changes as described in following section
+
+#### Flow
+
+```mermaid
+  sequenceDiagram
+    autonumber
+    Concentrator->>CUPS Server: POST /update-info
+    CUPS Server->>IoT Hub: Retrieve 'Concentrator' twin
+    CUPS Server->>CUPS Server: Verifies 'Concentrator' client certificate
+    Note right of CUPS Server: Logic for client certificate validation to be defined. First version including a 'cupsClientThumbprint' to verify against. The client certificate thumbprint should be reported as a property.
+    alt client certificate not valid
+        CUPS Server->>Concentrator: Return forbidden
+    else client certificate valid
+        alt no cups config available
+            CUPS Server->>Concentrator: Return no updates
```

Comment:
Is this an indication for malconfiguration? If the LBS tries to use CUPS and the configuration is not present, should we consider failing instead?

Reply:
You are right, good catch